sirepo/package_data/static/js/raydata.js

```diff
@@ -151,13 +413,43 @@ SIREPO.app.directive('appHeader', function(appState, panelState) {
 	    </div>
         `,
 	controller: function($scope) {
-	    $scope.haveScans = () => {
+	    $scope.haveScans = function() {
 		return ! $.isEmptyObject(appState.models.scans.selected);
 	    };
 	}
     };
 });
 
+SIREPO.app.directive('htmlAsString', function() {
+    return {
+        restrict: 'A',
+        scope: {
+	    html: '<',
+	    changeEvent: '@',
+	},
+        template: `
+            <element-to-replace></element-to-replace>
+        `,
+	controller: function($compile, $element, $timeout, $scope) {
+	    let currentElement = $element;
+	    function compile() {
+		$compile($scope.html)($scope , function(compiledContent){
+		    const r = angular.element(compiledContent);
+		    currentElement.replaceWith(r);
+		    currentElement = r;
+		});
+	    }
+	    $scope.$on($scope.changeEvent, function(){
+		// Need to wait one digest cycle before compile. The
```

Comment:
could we name this "callNextDigestCycle" or something? I don't like 0 timeouts, because it's an implementation detail that should be hidden.

Reply:
be69ce9e9